lib/contentProfileListener.js

```diff
@@ -98,16 +98,36 @@ class ContentProfileImporter extends SHRContentProfileParserListener {
 
     if (currentElement) {
       for (const name of names) {
+        let parentElements = this.getRecursiveBasedOns(currentElement.identifier);
+        
         if (name === 'Value' || name === '_Value') {
           path.push(new Identifier('', '_Value'));
 
-          if (currentElement.value && !(currentElement.value instanceof ChoiceValue)) {
+          // find value from self or most recent ancestor
+          let value = null;
+          for (const id of parentElements) {
+            const el = this._specs.dataElements.findByIdentifier(id);
+            value = el ? el.value : null;
+
+            if (value) {
+              break;
+            }
+          }
+
+          if (value && !(value instanceof ChoiceValue)) {
             currentElement = this._specs.dataElements.findByIdentifier(currentElement.value.effectiveIdentifier);
           } else {
             break; // Exit loop to reach error condition below
           }
         } else {
-          let value = [currentElement.value, ...currentElement.fields].find(field => {
+          // Collect fields from current elements and all parents
+          let fields = new Set();
+          for (const id of parentElements) {
+            const el = this._specs.dataElements.findByIdentifier(id);
+            [el.value, ...el.fields].forEach(field => { if (field) fields.add(field); });
```

Comment:
This works, so it's up to you if you want to change it, but... if a field is inherited, then it will be in the child class _and_ the parent class -- so you're adding it twice (although the child field may be a slightly different instance).  If the child field isn't a match (in the logic below), neither will the parent field be, so it's inefficient in that regard.

You could change the `Set` to a `Map` and use `field.identifier` as the key.  Then as you iterate, if the map already has the key just skip it (because you want the child field not the parent).  If it's a `ChoiceValue` then it won't have an `identifier` -- but since the value is the only thing that could have a choice, we know there is only one -- so you could use `'choice'` or even `'value'` as the key in that case (instead of the identifier).

Then, below, you iterate over the values of the Map.

Again, I think it works as-is, so it's up to you what you want to do.

Reply:
I fixed it as you suggested.